packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -321,6 +321,30 @@ class PouchLink extends CozyLink {
     return Boolean(this.indexes[name])
   }
 
+  /**
+   * Merge the selector and partial filter definitions.
+   *
+   * This is necessary because PouchDB does not support partial indexes, so
+   * we have to express any partial filter definition directly in the selector.
+   * Note the partial filter evaluation will be done in-memory, which might decrease
+   * performances.
+   *
+   * @param {object} selector - The query selector
+   * @param {object} partialFilter - The partial filter
+   * @returns {object} The merge between the selector and the partial filter
+   */
+  mergePartialIndexInSelector(selector, partialFilter) {
+    if (partialFilter) {
+      logger.info(
+        `PouchLink: The query contains a partial index but PouchDB does not support it. ` +
+          `Hence, the partial index definition is used in the selector for in-memory evaluation, ` +
+          `which might impact expected performances.`
+      )
```

Comment:
It's great to log it, but can the developer do something about this info? Do we recommend to not use partialFilter when using Pouch? Or do we ask the developer to donate / vote or something else?

Reply:
Ideally, for me the developer should not care about using Couch or Pouch when designing queries, so partialFilter should be used when it's possible. I guess we can suggest here the developer to at least tell PouchDB that this feature would be useful

Reply:
"plz sponsor my free time by donating on ... to implement it on PouchDB" :D